io/net/bonding.py

```diff
@@ -157,28 +163,57 @@ def ping_check(self, arg1):
         cmd = "ping -I %s %s -c 5"\
               % (self.bond_name, self.peer_first_ipinterface)
         if process.system(cmd, shell=True, ignore_status=True) != 0:
-            self.fail("ping failed in Mode %s, check bonding configuration"
-                      % arg1)
+            return False
+        return True
 
     def bond_fail(self, arg1):
         '''
         bond fail
         '''
         for interface in self.host_interfaces:
-            self.log.info("Failing interface %s for mode %s"
-                          % (interface, arg1))
+            if len(self.host_interfaces) > 1:
+                self.log.info("Failing interface %s for mode %s"
+                              % (interface, arg1))
+                cmd = "ifconfig %s down" % interface
+                if process.system(cmd, shell=True, ignore_status=True) != 0:
+                    self.fail("bonding not working when trying to down the\
+                               interface %s " % interface)
+                time.sleep(10)
+                if self.ping_check(arg1):
+                    self.log.info("Ping passed for Mode %s" % arg1)
+                else:
+                    self.fail("ping failed in Mode %s, check \
+                               bonding configuration" % arg1)
+                cmd = "cat /proc/net/bonding/%s" % self.bond_name
+                process.system_output(cmd, shell=True, verbose=True)
+                cmd = "ifconfig %s up" % interface
+                time.sleep(10)
+                if process.system(cmd, shell=True, ignore_status=True) != 0:
+                    self.fail("Not able to bring up the slave\
+                                    interface %s" % interface)
+                time.sleep(5)
+            else:
+                self.log.warning("Need a min of 2 host interfaces to test\
+                             slave failover in Bonding")
+
+        self.log.info("\n----------------------------------------")
+        self.log.info("Failing all interfaces for mode %s" % arg1)
+        self.log.info("----------------------------------------")
+        for interface in self.host_interfaces:
             cmd = "ifconfig %s down" % interface
             if process.system(cmd, shell=True, ignore_status=True) != 0:
-                self.fail("bonding not working when trying to down the\
-                          interface %s " % interface)
-            time.sleep(5)
-            self.ping_check(arg1)
-            cmd = "cat /proc/net/bonding/%s" % self.bond_name
-            process.system_output(cmd, shell=True, verbose=True)
+                self.fail("Could not bring down the interface %s " % interface)
+            time.sleep(10)
+        if not self.ping_check(arg1):
+            self.log.info("Ping to Bond interface failed. This is expected")
+        cmd = "cat /proc/net/bonding/%s" % self.bond_name
```

Comment:
`"/proc/net/bonding/%s" % self.bond_name` is used in multiple places.
Please save it in some variable.